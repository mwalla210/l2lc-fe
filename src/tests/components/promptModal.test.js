import React from 'react'
import renderer from 'react-test-renderer'
import PromptModal from '../../components/promptModal'
jest.mock('reactstrap', () => ({
  Modal: 'Modal',
  ModalBody: 'ModalBody',
  ModalHeader: 'ModalHeader',
  ModalFooter: 'ModalFooter',
}))
jest.mock('../../components/buttonDefault')

const defaultOptions = {
  closeFn: jest.fn(),
  denyOnClick: jest.fn(),
  confirmOnClick: jest.fn(),
  primaryButtonText: 'primaryButtonText',
  confirmClass: 'confirmClassName',
  headerClass: 'headerClassName',
  titleClass: 'titleClassName',
  titleImage: 'imageFilePath',
  open: true,
  title: 'title',
  content: 'content',
  }

  const alternateOptions = {
    closeFn: jest.fn(),
    confirmOnClick: jest.fn(),
    primaryButtonText: null,
    open: true,
    title: 'title',
    content: 'content',
    }

describe('PromptModal', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <PromptModal {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders with snapshot, alternate arguements', () => {
    let options = Object.assign({}, alternateOptions)
    const component = renderer.create(
      <PromptModal {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders and calls hide', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <PromptModal {...options}/>,
    )
    const inst = component.getInstance()

    expect(inst.props.closeFn.mock.calls.length).toBe(0)
    inst.hide()
    expect(inst.props.closeFn.mock.calls.length).toBe(1)
  })

  it('Renders and calls deny', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <PromptModal {...options}/>,
    )
    const inst = component.getInstance()

    expect(inst.props.denyOnClick.mock.calls.length).toBe(0)
    inst.deny()
    expect(inst.props.denyOnClick.mock.calls.length).toBe(1)
  })

  it('Renders and calls confirm', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <PromptModal {...options}/>,
    )
    const inst = component.getInstance()

    expect(inst.props.confirmOnClick.mock.calls.length).toBe(0)
    inst.confirm()
    expect(inst.props.confirmOnClick.mock.calls.length).toBe(1)
  })

})
