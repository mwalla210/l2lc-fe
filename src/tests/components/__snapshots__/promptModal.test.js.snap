// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromptModal Renders with snapshot 1`] = `
<Modal
  isOpen={true}
  size="small"
  toggle={[Function]}
>
  <ModalHeader
    className="titleClassName"
    toggle={[Function]}
  >
    <img
      alt="imageFilePath"
      src="../../style/open-iconic-master/svg/imageFilePath.svg"
      style={
        Object {
          "marginRight": "3px",
          "marginTop": "-2px",
          "width": "14px",
        }
      }
    />
    title
  </ModalHeader>
  <ModalBody>
    content
  </ModalBody>
  <ModalFooter />
</Modal>
`;

exports[`PromptModal Renders with snapshot, alternate arguements 1`] = `
<Modal
  isOpen={true}
  size="small"
  toggle={[Function]}
>
  <ModalHeader
    toggle={[Function]}
  >
    title
  </ModalHeader>
  <ModalBody>
    content
  </ModalBody>
  <ModalFooter />
</Modal>
`;
